creusot-contracts/src/std/option.rs

```diff
@@ -1,25 +1,213 @@
 use crate as creusot_contracts;
-use crate::std::default::DefaultSpec;
+use crate::std::{default::DefaultSpec, fun::FnOnceSpec};
 use creusot_contracts_proc::*;
 
 extern_spec! {
     mod std {
         mod option {
             impl<T> Option<T> {
+                #[ensures(result == (*self != None))]
+                fn is_some(&self) -> bool;
+
                 #[ensures(result == (*self == None))]
                 fn is_none(&self) -> bool;
 
                 #[requires(self != None)]
                 #[ensures(Some(result) == self)]
                 fn unwrap(self) -> T;
 
+                #[requires(self != None)]
+                #[ensures(Some(result) == self)]
+                fn expect(self, msg: &str) -> T;
+
+                #[ensures(result == match self {
+                    Some(inner) => inner,
+                    None => default,
+                })]
```

Comment:
Pattern matching are usually a bad idea in post-conditions, they are badly handled by provers. It's usually better to have two post-conditions, one for None and one for Some (with no quantifier or an existential quantifier if possible, or a forall quantifier.).